From 1d307ed7f268d3bb511985c18d580ab8a104ed41 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@redhat.com>
Date: Fri, 28 Nov 2008 09:34:56 +1000
Subject: [PATCH] Xi: remove the GetExtensionVersion hack - don't reply with supported version.

XI2 abuses the GEV request to reply with the min/major version of the
supported extension if the length for the name is 0. Don't do that, yet.

Signed-off-by: Peter Hutterer <peter.hutterer@redhat.com>
---
 Xi/getvers.c |   13 -------------
 1 files changed, 0 insertions(+), 13 deletions(-)

diff --git a/Xi/getvers.c b/Xi/getvers.c
index 43a1a4d..88ff199 100644
--- a/Xi/getvers.c
+++ b/Xi/getvers.c
@@ -105,19 +105,6 @@ ProcXGetExtensionVersion(ClientPtr client)
 
     pXIClient = dixLookupPrivate(&client->devPrivates, XIClientPrivateKey);
 
-    /* GetExtensionVersionReq before XI 2 didn't supply the client's
-     * major/minor. So we don't actually have a clue what they support.
-     * {major|minor}Version was added as part of XI, so if they are set, we
-     * know we can trust it. In this case the client must set nbytes to 0
-     * though, otherwise we have to assume that the version are padding
-     * garbage.
-     */
-    if (!stuff->nbytes) /* Client using XQueryInputVersion(). */
-    {
-        pXIClient->major_version = stuff->majorVersion;
-        pXIClient->minor_version = stuff->minorVersion;
-    } /* else version unknown, leave it at 0.0 */
-
     rep.repType = X_Reply;
     rep.RepType = X_GetExtensionVersion;
     rep.length = 0;
-- 
1.6.0.4

